platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -525,6 +536,37 @@ protected BundleContext getContext() {
     }
   }
 
+  @Override
+  public void restart() {
+    try {
+      if (!restartServiceWrapperIfControlled()) {
+        // create the restart.jvm file such that we would rely on the ddf script to restart
+        // ourselves
+        // and not on karaf. This will have the advantage to restart anything else (e.g. solr) that
+        // is also managed by that script
+        LOGGER.debug("generating restart.jvm file");
+        AccessController.doPrivileged(
+            (PrivilegedExceptionAction<Void>)
+                () -> {
+                  FileUtils.touch(
+                      Paths.get(System.getProperty("ddf.home"), "bin", "restart.jvm").toFile());
+                  // make sure Karaf is not going to restart us as we want the ddf script to do it
+                  System.setProperty("karaf.restart.jvm", "false");
+                  systemService.halt();
+                  return null;
+                });
+      }
+      SecurityLogger.audit("Restarting system");
+      LOGGER.info("Restarting the system.");
+    } catch (PrivilegedActionException e) {
+      SecurityLogger.audit("Failed to restart system");
+      LOGGER.debug("failed to request a restart: ", e.getException());
+    } catch (Exception e) {
```

Comment:
It is really meant to be a catch all so I would rather want to keep it as is

Reply:
The risk of catching `Exception` here is that if the code inside the `try` block changes and a new checked exception gets thrown, the person making the change can easily miss it and not handle it the way it should be handled, e.g., send something to the audit log.

Reply:
That is exactly my point, since the exception would fail the restart, I want to catch it and handle it as done here. This is no different then catching the `PrivilegedActionException which would piggy back any exceptions thrown from within the action.

Reply:
What other checked exceptions can be thrown within the `try` block? What would be the harm in the last catch block being `catch (OtherException1 | OtherException2 | RuntimeException)` instead of `catch (Exception)`?

Reply:
It ain’t a matter of harm … it is a matter to define a catch all here on purpose to not let anything bubble out to Jolokia. This is one of those case where it is meant o literally not let anything other bubble out.